### PR TITLE
zh-cn: init translation of svelte components

### PR DESCRIPTION
**Pull request by hanyujie2002**
### Description

Same as title
### Motivation

Ditto
### Additional details

None
### Related issues and pull requests

Relates to #16965

### CODE REVIEW

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+     </li>
+     {:else}
+     <li>Nothing to do here!</li>
+     {/each}
+   </ul>
+   ```
+
+   待办事项列表将显示在页面上，复选框应该可以正常工作（尝试勾选/取消勾选几个，然后观察过滤器是否按预期工作），但是我们的“x out of y items completed”状态标题将不再相应地更新。这是因为我们的 `Todo` 组件通过属性接收待办事项，但没有向其父组件发送任何信息。我们将在以后修复这个问题。
+
+## 在组件之间共享数据：props-down、events-up 模式
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:261]: How about translate this into (ref: [vue docs](https://v2.cn.vuejs.org/v2/style-guide/#%E9%9A%90%E6%80%A7%E7%9A%84%E7%88%B6%E5%AD%90%E7%BB%84%E4%BB%B6%E9%80%9A%E4%BF%A1%E8%B0%A8%E6%85%8E%E4%BD%BF%E7%94%A8)):

```plain
属性向下传递、事件向上传递
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+
+在 Svelte 中，应用程序由一个或多个组件组成。组件是可重用的、自包含的代码块，将 HTML、CSS 和 JavaScript 组合在一起，写入 `.svelte` 文件中。组件可以大或小，但通常明确定义：最有效的组件只服务于某个明显的目的。
+
+定义组件的好处类似于将代码组织成可管理的片段的更一般的最佳实践。它将帮助你理解它们之间的关系，促进重用，并使你的代码更易于理解、维护和扩展。
+
+但是，你如何知道什么应该拆分为组件呢？
+
+这没有硬性规定。有些人更喜欢直观的方法，开始查看标记并在每个似乎具有自己逻辑的组件和子组件周围画框。
+
+其他人应用同样用于决定是否创建新函数或对象的技术。其中一种技术是单一职责原则——也就是说，一个组件理想情况下只做一件事。如果它变得越来越庞大，应该将其拆分为更小的子组件。
```
[the remark this answers is not in the document]
Comment (hanyujie2002) [reply files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:80]: Sorry, but I can't agree with this recommendation. The original content is "Other people apply the same techniques used for deciding if you should create a new function or object." In my opinion, the "其他人应使用同样的技术来决定是否创建新函数或对象" seems to be more suitable if it is in the article introducing how to create new function or object. @yin1999

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md (hunk 1 above)
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:261]:
```suggestion
## 在组件之间共享数据：属性向下传递、事件向上传递模式
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+   {#each filterTodos(filter, todos) as todo (todo.id)}
+   <li class="todo">
+     <Todo {todo} on:update={(e) => updateTodo(e.detail)} on:remove={(e) =>
+     removeTodo(e.detail)} />
+   </li>
+   ```
+
+4. 再次尝试你的应用程序，你将看到你可以删除、添加、编辑、取消编辑和切换待办事项的完成状态。当待办事项完成时，“x out of y items completed”的状态标题将相应更新。
+
+正如你所见，在 Svelte 中实现“props-down，events-up”模式很容易。尽管如此，对于简单的组件，`bind` 可能是一个不错的选择；Svelte 允许你自由选择。
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:517]:
```suggestion
正如你所见，在 Svelte 中实现“属性向下传递、事件向上传递”模式很容易。尽管如此，对于简单的组件，`bind` 可能是一个不错的选择；Svelte 允许你自由选择。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+
+现在，我们已经完成了应用程序所需的所有功能。我们可以显示、添加、编辑和删除待办事项，将它们标记为已完成，并按状态进行筛选。
+
+在本文中，我们涵盖了以下主题：
+
+- 将功能提取到新组件中
+- 使用作为属性接收到的处理程序将信息从子组件传递到父组件
+- 使用 `bind` 指令将信息从子组件传递到父组件
+- 使用 `if` 块有条件地渲染标记块
+- 实现“props-down, events-up”通信模式
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:555]:
```suggestion
- 实现“属性向下传递、事件向上传递”通信模式
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+     </li>
+     {:else}
+     <li>Nothing to do here!</li>
+     {/each}
+   </ul>
+   ```
+
+   待办事项列表将显示在页面上，复选框应该可以正常工作（尝试勾选/取消勾选几个，然后观察过滤器是否按预期工作），但是我们的“x out of y items completed”状态标题将不再相应地更新。这是因为我们的 `Todo` 组件通过属性接收待办事项，但没有向其父组件发送任何信息。我们将在以后修复这个问题。
+
+## 在组件之间共享数据：props-down、events-up 模式
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:261]:
```suggestion
## 在组件之间共享数据：属性向下传递、事件向上传播模式
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 6]
```diff
@@ -0,0 +1,560 @@
...
+   {#each filterTodos(filter, todos) as todo (todo.id)}
+   <li class="todo">
+     <Todo {todo} on:update={(e) => updateTodo(e.detail)} on:remove={(e) =>
+     removeTodo(e.detail)} />
+   </li>
+   ```
+
+4. 再次尝试你的应用程序，你将看到你可以删除、添加、编辑、取消编辑和切换待办事项的完成状态。当待办事项完成时，“x out of y items completed”的状态标题将相应更新。
+
+正如你所见，在 Svelte 中实现“props-down，events-up”模式很容易。尽管如此，对于简单的组件，`bind` 可能是一个不错的选择；Svelte 允许你自由选择。
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:517]:
```suggestion
正如你所见，在 Svelte 中实现“属性向下传递、事件向上传播”模式很容易。尽管如此，对于简单的组件，`bind` 可能是一个不错的选择；Svelte 允许你自由选择。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 7]
```diff
@@ -0,0 +1,560 @@
...
+
+现在，我们已经完成了应用程序所需的所有功能。我们可以显示、添加、编辑和删除待办事项，将它们标记为已完成，并按状态进行筛选。
+
+在本文中，我们涵盖了以下主题：
+
+- 将功能提取到新组件中
+- 使用作为属性接收到的处理程序将信息从子组件传递到父组件
+- 使用 `bind` 指令将信息从子组件传递到父组件
+- 使用 `if` 块有条件地渲染标记块
+- 实现“props-down, events-up”通信模式
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:555]:
```suggestion
- 实现“属性向下传递、事件向上传播”通信模式
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md (hunk 1 above)
Comment (hanyujie2002) [reply files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md]: How about "属性向下传递、事件向上传播"？I prefer this to that old translation 🤔 @yin1999

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 8]
```diff
@@ -0,0 +1,560 @@
...
+
+在 Svelte 中，应用程序由一个或多个组件组成。组件是可重用的、自包含的代码块，将 HTML、CSS 和 JavaScript 组合在一起，写入 `.svelte` 文件中。组件可大可小，但通常具有明确定义：最有效的组件只服务于某个明确的目的。
+
+定义组件的好处类似于将代码组织成可管理的片段的更一般的最佳实践。它将帮助你理解它们之间的关系，促进重用，并使你的代码更易于理解、维护和扩展。
+
+但是，你如何知道什么应该拆分为组件呢？
+
+这没有硬性规定。有些人更喜欢直观的方法，开始查看标记并在每个似乎具有自己逻辑的组件和子组件周围绘制封装。
+
+其他人应用同样用于决定是否创建新函数或对象的技术。其中一种技术是单一职责原则——也就是说，一个组件理想情况下只做一件事。如果它变得越来越庞大，应该将其拆分为更小的子组件。
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:80]:
```suggestion
其他人借鉴了一些用于创建新函数或对象的技术。其中一种技术是单一职责原则——也就是说，一个组件理想情况下只做一件事。如果它变得越来越庞大，应该将其拆分为更小的子组件。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 9]
```diff
@@ -0,0 +1,560 @@
...
+     <li>Nothing to do here!</li>
+     {/each}
+   </ul>
+   ```
+
+   待办事项列表将显示在页面上，复选框应该可以正常工作（尝试勾选/取消勾选几个，然后观察过滤器是否按预期工作），但是我们的“x out of y items completed”状态标题将不再相应地更新。这是因为我们的 `Todo` 组件通过属性接收待办事项，但没有向其父组件发送任何信息。我们将在以后修复这个问题。
+
+## 在组件之间共享数据：属性向下传递、事件向上传播模式
+
+`bind` 指令非常简单直观，可以让你在父组件和子组件之间共享数据。然而，当应用程序变得更大、更复杂时，便难以跟踪所有绑定的值。另一种方法是使用“属性向下传递，事件向上传播”通信模式。
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:263]:
```suggestion
`bind` 指令非常简单直观，可以让你在父组件和子组件之间共享数据。然而，当应用程序变得更大、更复杂时，便难以跟踪所有绑定的值。另一种方法是使用“属性向下传递、事件向上传播”通信模式。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:80]: :+1:

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md (hunk 1 above)
Comment (yin1999) [reply files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md]: +1

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 10]
```diff
@@ -0,0 +1,560 @@
...
+   }
+   ```
+
+### 更新标记
+
+现在我们需要更新 `Todo` 组件的标记，以在适当的操作发生时调用上述函数。
+
+为了处理编辑模式，我们使用了布尔类型的 `editing` 变量。当它为 `true` 时，它应该显示用于编辑待办事项名称的 `<input>` 字段以及 _Cancel_ 和 _Save_ 按钮。当它不处于编辑模式时，它将显示复选框、待办事项名称以及编辑和删除待办事项的按钮。
+
+为了实现这一点，我们将使用 [`if` 块](https://svelte.dev/docs#if)。`if` 块会根据条件渲染标记。请注意，它不仅仅是根据条件显示或隐藏标记——它会根据条件在 DOM 中动态添加和移除元素。
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:369]:
```suggestion
为了实现这一点，我们将使用 [`if` 块](https://svelte.dev/docs/logic-blocks#if)。`if` 块会根据条件渲染标记。请注意，它不仅仅是根据条件显示或隐藏标记——它会根据条件在 DOM 中动态添加和移除元素。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 11]
```diff
@@ -0,0 +1,560 @@
...
+     <Todo {todo} on:update={(e) => updateTodo(e.detail)} on:remove={(e) =>
+     removeTodo(e.detail)} />
+   </li>
+   ```
+
+4. 再次尝试你的应用程序，你将看到你可以删除、添加、编辑、取消编辑和切换待办事项的完成状态。当待办事项完成时，“x out of y items completed”的状态标题将相应更新。
+
+正如你所见，在 Svelte 中实现“属性向下传递、事件向上传播”模式很容易。尽管如此，对于简单的组件，`bind` 可能是一个不错的选择；Svelte 允许你自由选择。
+
+> **备注：** Svelte 提供了更高级的机制来在组件之间共享信息：[Context API](https://svelte.dev/docs#setContext) 和 [Stores](https://svelte.dev/docs#svelte_store)。Context API 提供了一种机制，使组件及其后代组件能够在不传递数据和函数作为属性或分发大量事件的情况下进行通信。Stores 允许你在不具有层次关系的组件之间共享响应式数据。我们将在本系列的后续内容中介绍 Stores。
```
Comment (hanyujie2002) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:519]:
```suggestion
> **备注：** Svelte 提供了更高级的机制来在组件之间共享信息：[Context API](https://svelte.dev/docs/svelte#setcontext) 和 [Stores](https://svelte.dev/docs/svelte-store)。Context API 提供了一种机制，使组件及其后代组件能够在不传递数据和函数作为属性或分发大量事件的情况下进行通信。Stores 允许你在不具有层次关系的组件之间共享响应式数据。我们将在本系列的后续内容中介绍 Stores。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 12]
```diff
@@ -0,0 +1,560 @@
...
+     <Todo {todo} on:update={(e) => updateTodo(e.detail)} on:remove={(e) =>
+     removeTodo(e.detail)} />
+   </li>
+   ```
+
+4. 再次尝试你的应用程序，你将看到你可以删除、添加、编辑、取消编辑和切换待办事项的完成状态。当待办事项完成时，“x out of y items completed”的状态标题将相应更新。
+
+正如你所见，在 Svelte 中实现“属性向下传递、事件向上传播”模式很容易。尽管如此，对于简单的组件，`bind` 可能是一个不错的选择；Svelte 允许你自由选择。
+
+> **备注：** Svelte 提供了更高级的机制来在组件之间共享信息：[Context API](https://svelte.dev/docs/svelte#setcontext) 和 [Stores](https://svelte.dev/docs/svelte-store)。Context API 提供了一种机制，使组件及其后代组件能够在不传递数据和函数作为属性或分发大量事件的情况下进行通信。Stores 允许你在不具有层次关系的组件之间共享响应式数据。我们将在本系列的后续内容中介绍 Stores。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:519]:
```suggestion
> **备注：** Svelte 提供了更高级的机制来在组件之间共享信息：[Context API](https://svelte.dev/docs/svelte#setcontext) 和 [Store](https://svelte.dev/docs/svelte-store)。Context API 提供了一种机制，使组件及其后代组件能够在不传递数据和函数作为属性或分发大量事件的情况下进行通信。Store 允许你在不具有层次关系的组件之间共享响应式数据。我们将在本系列的后续内容中介绍 Store。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 13]
```diff
@@ -0,0 +1,560 @@
+---
+title: 将我们的 Svelte 应用组件化
+slug: Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_components
+---
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:4]: Let's also add the [`sourceCommit`](https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md#%E5%85%83%E6%95%B0%E6%8D%AE)

```suggestion
l10n:
  sourceCommit: e9be22eaa7416206e3f263a058f0d509a7f81f88
---
```

**Review by yin1999 — APPROVED** [review]
Thank you @hanyujie2002 :+1:

**Pull request merged**